lib/media/streaming_engine.js

```diff
@@ -1987,7 +1987,7 @@ shaka.media.StreamingEngine = class {
       const isResetMediaSourceNecessary =
           mediaState.lastCodecs && mediaState.lastMimeType &&
           this.playerInterface_.mediaSourceEngine.isResetMediaSourceNecessary(
-              mediaState.type, mediaState.stream, mimeType, codecs);
+              mediaState.type, mediaState.stream, mimeType, fullCodecs);
```

Comment:
I'm afraid it will let us reset media source whenever profile changes (i.e. `mp4a.40.2 -> mp4a.40.5`) and it's not the option. Lots of platforms do not support `SourceBuffer.changeType()` API so it would make us reload `MediaSource` a lot.

Reply:
I agree, we cannot accept this change.

Reply:
```
    if (!shaka.media.Capabilities.isTypeSupported(newMimeType) ||
        (!this.sequenceMode_ &&
        shaka.util.MimeUtils.RAW_FORMATS.includes(newMimeType))) {
      needTransmux = true;
    }
```
@avelad @tykus160 
I understand your concern. The problem for us is the condition of this if statement in `isCodecSwitchNecessary_` in MediaSourceEngine. Calling `isTypeSupported` here with mimeType that has truncated base codec results in a video playback failure and an app crash on Comcast set top boxes. 

How about limiting this change to Comcast platforms only and leave everything else as is?

Reply:
I've been looking at this with @agajassi. What @tykus160 described as a potential issue is already happening and part of what we are trying to fix.
`isResetMediaSourceNecessary` depends on the return of `isCodecSwitchNecessary_` in `MediaSourceEngine`. Because a truncated/base codec is being passed to `isCodecSwitchNecessary_`, when `isTypeSupported` is evaluated, it's checking an incomplete mimeType (i.e `'audio/mp4; codecs="mp4a"'` vs `'audio/mp4; codecs="mp4a.40.2"'`). Calling `MediaSource.isTypeSupported('audio/mp4; codecs="mp4a"')` returns `false` on all browsers and in the case of Comcast causes the browser to crash. That's why we need the full codec string passed down to `isCodecSwitchNecessary_`. 

Logs showing current behavior
![codec-switch](https://github.com/user-attachments/assets/cd24c3d0-434c-4c0b-8b7b-9a746bb3ac2e)

Logs with fix
![codec-switch-fix](https://github.com/user-attachments/assets/44f56710-ec65-4cfb-805b-765d43bfb278)



The full codecs will get normalized in `isCodecSwitchNecessary_` before evaluating if the codecs have changed, so this change looks safe to me. But if passing the full codec strings may cause other problems we're not aware of, is there another way we can access the full codec string of the new stream in `isCodecSwitchNecessary_`?

Reply:
> Calling `MediaSource.isTypeSupported('audio/mp4; codecs="mp4a"')` returns `false` on all browsers and in the case of Comcast causes the browser to crash.

I would argue that this is the true root of the problem.  You shouldn't be able to crash the browser by calling a function.

How about we patch isTypeSupported on this particular Comcast device?

Reply:
I've updated the logic by moving it to `isCodecSwitchNecessary_` in the MediaSourceEngine class. This update is specifically targeted for the Comcast platform and is limited to use cases with `avc` codecs without a profile. Let me know what your thoughts are on this.

Reply:
@agajassi after taking another look it seems shape of this PR in commit https://github.com/shaka-project/shaka-player/pull/7233/commits/7229c3b8bf3cb2f33459a93b12f25748317f0b1f was suffcient. Could you please revert 2 next commits so we could merge it? 🙂

Reply:
Ok thanks @tykus160 sounds good